File: src/main/java/back/ecommerce/order/service/OrderGroupDto.java
package back.ecommerce.order.service;

import back.ecommerce.order.entity.OrderGroup;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public class OrderGroupDto {
	private final String orderCode;
	private final String name;
	private final Integer quantity;
	private final Long totalPrice;

	public static OrderGroupDto create(OrderGroup orderGroup) {
		return new OrderGroupDto(orderGroup.getOrderCode(), orderGroup.getName(), orderGroup.getQuantity(),
			orderGroup.getTotalPrice());
	}
}

File: src/main/java/back/ecommerce/order/service/OrderGroupDto.kt
package back.ecommerce.order.service

import back.ecommerce.order.entity.OrderGroup

data class OrderGroupDto(
    val orderCode: String,
    val name: String,
    val quantity: Int,
    val totalPrice: Long
) {
    companion object {
        fun create(orderGroup: OrderGroup): OrderGroupDto {
            return OrderGroupDto(
                orderGroup.orderCode, orderGroup.name, orderGroup.quantity,
                orderGroup.totalPrice
            )
        }
    }
}
